### DISCUSSION TITLE
Potential bug in v0.109.x when root_path="/"

### DISCUSSION THREAD

**Discussion by dbertero-w** [fastapi/fastapi#11138] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py

from fastapi import FastAPI

app = FastAPI(
    title="title",
    version="0.0.1",
    description="Description",
    root_path="/",
)

@app.get("/path")
def path() -> str:
    return "Success"


# test_main.py
from fastapi.testclient import TestClient

def test_path():
    with TestClient(app) as client:
        response = client.get("/path")

    assert response.status_code == 200. # This fails, yields 404


def test_path_bis():
    with TestClient(app) as client:
        response = client.get("//path")

    assert response.status_code == 200. # This works
```

### Description

* In version 0.109.x, when testing, if `root_path="/"`, an extra slash is needed in the path. This did not happen in v0.108 and prior.

### Operating System

Linux, macOS

### FastAPI Version

v0.109.1

### Pydantic Version

2.5.2

### Python Version

3.10.13

**Comment by Leon0824** [comment]
I got a similar issue. They may be caused by the same root issue.

## Example code

```py
from fastapi import FastAPI, status


fastapi_app = FastAPI(
    root_path='/api' # Notice the path is named 'api'
)


@fastapi_app.get('/hi') # Notice the path is not named with 'api' 
def hi(): return 'Hi'


@fastapi_app.post('/api-key/generate') # Notice the path is named 'api-xxx'
def generate_api_key(): return 'keykeykey'
```

## Test

```py
from http import HTTPStatus

from fastapi.testclient import TestClient


def test_hi(self, client: TestClient):
    with client: response = client.get(url='/hi')
    assert response.status_code == HTTPStatus.OK # This test is pass


def test_generate_api_key(self, client: TestClient):
    with client: response = client.post(url='/api-key/generate')
    assert response.status_code == HTTPStatus.CREATED # This test is 404
```

As long as set `root_path` to `/api`, any endpoints named 'apixxx' are all 404.

Like @dbertero-w 's case, We can access /api-key/generate with additional '/api' prefix:

```py
def test_generate_api_key(self, client: TestClient):
    with client: response = client.post(url='/api/api-key/generate') # Add additional '/api' prefix
    assert response.status_code == HTTPStatus.CREATED # This test is pass
```

**Comment by YuriiMotov** [comment, marked as answer]
Both of mentioned problems have already been fixed.
Works well in latest version
